gcpy/file_regrid.py

```diff
@@ -448,157 +551,457 @@ def regrid_ll_to_cssg(
+        )
+
+        # Set the lev:positive attribute accordingly
+        dset = flip_lev_coord_if_necessary(
+            dset,
+            dim_format_in="classic",
+            dim_format_out="classic"
+        )
+
+        # Save lat/lon coordinate metadata
+        dset = save_ll_metadata(
+            dset,
+            verbose=verbose
         )
 
-    return ds_out
+    return dset
+
+
+def flip_lev_coord_if_necessary(
+        dset,
+        dim_format_in,
+        dim_format_out
+):
+    """
+    Determines whether it is necessary to flip the "lev" and "ilev"
+    coords of an xarray.Dataset in the vertical depending on the
+    values ofdim_format_in and dim_format_out.  Also sets the
+    attributes "lev:positive" and "ilev:positive" accordingly.
+
+    Args:
+    -----
+    dset : xarray.Dataset
+        The input dataset.
+    dim_format_in : str
+        Input grid format ("classic", "checkpoint", "diagnostic").
+    dim_format_out : str
+        Output grid format ("classic", "checkpoint", "diagnostic").
+
+    Args:
+    -----
+    dset : xarray.Dataset
+        The modified dataset.
+
+    Remarks:
+    --------
+    (1) classic    : lev is in ascending order  (lev:positive="up"  )
+    (2) diagnostic : lev is in ascending order* (lev:positive="up"  )
+    (3) checkpoint : lev is in descending order (lev:positive="down")
+
+    *Except for the Emissions collection, which has lev arranged
+     in descending order.
+    """
+    verify_variable_type(dset, xr.Dataset)
+    verify_variable_type(dim_format_in, str)
+    verify_variable_type(dim_format_out, str)
+
+    # ==================================================================
+    # checkpoint/diagnostic to classic: lev, ilev in ascending order
+    # ==================================================================
+    if dim_format_in != "classic" and dim_format_out == "classic":
+
+        # Flip lev and set to eta values at midpoints (if necessary)
+        if "ilev" in dset.coords:
+            dset = dset.reindex(ilev=dset.ilev[::-1])
+            if any(var > 1.0 for var in dset.ilev):
+                coord = get_ilev_coord(
+                    n_lev=dset.dims["ilev"],
+                    top_down=False
+                )
+                dset = dset.assign_coords({"ilev": coord})
+            dset.ilev.attrs["positive"] = "up"
+
+        # Flip lev and set to eta values at midpoints (if necessary)
+        if "lev" in dset.coords:
+            dset = dset.reindex(lev=dset.lev[::-1])
+            if any(var > 1.0 for var in dset.lev):
+                coord = get_lev_coord(
+                    n_lev=dset.dims["lev"],
+                    top_down=False
+                )
+                dset = dset.assign_coords({"lev": coord})
+            dset.lev.attrs["positive"] = "up"
+
+        return dset
+
+    # ==================================================================
+    # classic/diagnostic to checkpoint: lev in descending order
```

Comment:
Put a TODO here about the emissions diagnostic?

Reply:
Now updated in commit 922f233.